Update CHANGELOG.md for the 202011.00 release. (#1359)
aws/aws-iot-device-sdk-embedded-C: CHANGELOG.md

# Changelog for AWS IoT Device SDK for Embedded C

## 202011.00 (November 2020)

### Major Changes

> - This release introduces the re-factored **HTTP** client library, **AWS IoT Device Defender** service client library, and **AWS IoT Jobs** service library.

Reviewer: ```suggestion
- This release adds the re-factored **HTTP** client library, **AWS IoT Device Defender** service client library, and **AWS IoT Jobs** service library to the list of supported C-SDK libraries.
```

> These libraries have gone through code quality checks including verification that no function has a [GNU Complexity](https://www.gnu.org/software/complexity/manual/complexity.html) score greater than 8, and checks against deviations from the mandatory rules in the [MISRA coding standard](https://www.misra.org.uk/MISRAHome/MISRAC2012/tabid/196/Default.aspx). These libraries have also undergone both static code analysis from [Coverity statical analysis](https://scan.coverity.com/) and validation of memory safety through the [CBMC automated reasoning tool](https://www.youtube.com/watch?v=YwQHAPRhQkI&feature=youtu.be&t=1721).

Reviewer 2: Do we need to post the CBMC video? Or should we use the link: http://www.cs.cmu.edu/~modelcheck/cbmc/

Author: We can use that link instead

> - This release uses submodule references to the following new individual repositories for the HTTP, AWS IoT Defender service client, and AWS IoT Jobs service libraries under the `libraries` folder:

Reviewer: ```suggestion
- This release uses submodule references to the following new individual repositories for the HTTP, AWS IoT Device Defender service, and AWS IoT Jobs service client libraries under the `libraries` folder:
```

Author: Not sure if client should follow jobs

   - [FreeRTOS/coreHTTP](https://github.com/FreeRTOS/coreHTTP) for the HTTP client library

>    - [aws/device-defender-for-aws-iot-embedded-sdk](https://github.com/aws/device-defender-for-aws-iot-embedded-sdk) for the AWS IoT Device Defender service client library.

Reviewer: ```suggestion
   - [aws/device-defender-for-aws-iot-embedded-sdk](https://github.com/aws/device-defender-for-aws-iot-embedded-sdk) for the AWS IoT Device Defender client library.
```
